### PR TITLE
Filter out record side names from Bandcamp

### LINKED ISSUE #3093: Bandcamp connector - filter out record side names
**Is your feature request related to a problem? Please describe.**
Some records on bandcamp have the side and track number in the track name. This is not part of the track name, but is only used to indicate where on the record the track is located. Since they're not part of the track name, I want to scrobble the track without it.

**Describe the solution you'd like**
It would be nice if the connector could detect the usage of these indicators and remove them from the track name.

**Describe alternatives you've considered**
Manually editing the track name is possible, but it requires some extra work.

**Additional context**
Some examples below:
image

image

Examples on Bandcamp: 
- https://zementrec.bandcamp.com/album/002
- https://hectoroaks.bandcamp.com/album/neanderthal-disco

Note that it is possible for bandcamp listings to go further then A and B in case of multiple records in 1 listing (for example C1, C2, D1 etc)

### PR DESCRIPTION
**Pull request by dyerai**
**Describe the changes you made**
Checks for a record side prefix in the track title on the album page of the Bandcamp connector. If the prefix exists, it's filtered out of the track title string. This allows for more accurate scrobbles of Bandcamp tracks

**Additional context**
Closes #3093

### CODE REVIEW

**Comment by jaccarmac** [comment]
Looks pretty good. I think it would be ideal if there were a check of all tracks on a page, similar to what happens just below your change: https://github.com/web-scrobbler/web-scrobbler/blob/95d657f91b0cd14b30167a1e82012ff337d4b8de/src/connectors/bandcamp.js#L70

File: src/core/content/util.js [hunk 1]
```diff
@@ -110,6 +110,16 @@ const Util = {
 		return { artist, track };
 	},
 
+	/**
+	 * Remove record side from track title.
+	 * @param {String} str String contains track title
+	 * @return {String} String contains track title without record side
+	 */
+	removeRecordSide(str) {
```
Comment (inverse) [inline src/core/content/util.js:118]: Could we include a test case for this?

See https://github.com/web-scrobbler/w`eb-scrobbler/blob/3ca9d78037931c8d5517d08fde3b1f5d0330036d/tests/content/util.js#L23 How this is setup for `splitArtistTrack

Comment (inverse) [inline src/core/content/util.js:118]: Lets add a test case for this.

See: https://github.com/web-scrobbler/web-scrobbler/blob/3ca9d78037931c8d5517d08fde3b1f5d0330036d/tests/content/util.js#L23 for how we do it for `splitArtistTrack

File: src/connectors/bandcamp.js [hunk 2]
```diff
@@ -258,6 +262,22 @@ function isArtistVarious(artist, track) {
 	return false;
 }
 
+function trackContainsRecordSide() {
+	const RECORD_SIDE_REGEX = /^[A-Z][1-9]\.? /;
```
Comment (inverse) [inline src/connectors/bandcamp.js:266]: This could come from `Util`  so we have a single regex definiton

File: src/core/content/util.js (hunk 1 above)
Comment (dyerai) [reply src/core/content/util.js:124]: Added some test cases, let me know if you can think of anything more.

**Review by inverse — APPROVED** [review]
Amazing thanks 👍🏼

**Pull request merged**